src/separator/io.clj

```diff
@@ -105,7 +105,10 @@
           (if-let [headers @state]
             (if (parse-error? row)
               (rf acc row)
-              (rf acc (zipmap headers row)))
+              (let [row-meta (meta row)
+                    record (zipmap headers row)
+                    record-with-meta (with-meta record row-meta)]
+                (rf acc record-with-meta)))
```

Comment:
⛳ This is simple enough to inline, but one level of let-binding is probably the most clear:
```suggestion
              (let [record (with-meta (zipmap headers row) (meta row))]
                (rf acc record)))
```